### PR TITLE
KMSDRM: Work in Progress DRM backend

### PR DESCRIPTION
**Pull request by JohnnyonFlame**
This is a set of Work in Progress patches to provide KMS/DRM support for SDL1.2.

# Tasks:
- [x] Initial prototype implementation
- [x] Support for creating dumb buffers from different Pixel Formats
- [x] Support for swizzled color formats
- [x] Support for double buffering
- [x] Support for triple buffering
- [x] Support for arbitrary refresh rates.
- [x] Keyboard Input support
- [x] Mouse Input support
- [x] Investigate if joystick code needs to be updated, and if so, discuss how to proceed
- [x] Refactor SDL_kmsdrmextras.h into a .c file.
- [x] Once work is nearly done, make the backend less chatty by refactoring printf's into a (disabled) debug printf macro

### CODE REVIEW

File: src/video/kmsdrm/SDL_kmsdrmvideo.c [hunk 1]
```diff
@@ -437,9 +481,33 @@ static void KMSDRM_UnlockHWSurface(_THIS, SDL_Surface *surface)
...
+	if (this->screen->flags & SDL_DOUBLEBUF) {
+		// drmModeObjectSetProperty seems to wait for VBlank already, this doesn't
+		// seem really necessary.
+		/*drmVBlank vb = {
+			.request.type = DRM_VBLANK_RELATIVE,
+			.request.sequence = 1,
+		};
+
+		drmWaitVBlank(drm_fd, &vb);*/
+		drmModeObjectSetProperty(drm_fd, drm_active_pipe->plane,
```
Comment (JohnnyonFlame) [inline src/video/kmsdrm/SDL_kmsdrmvideo.c]: setting "FB_ID" on the plane here is a blocking operation that waits for vblank, is this expected behavior?

File: include/SDL_video.h [hunk 2]
```diff
@@ -144,6 +144,7 @@ typedef struct SDL_Surface {
 #define SDL_OPENGLBLIT	0x0000000A	/**< Create an OpenGL rendering context and use it for blitting */
 #define SDL_RESIZABLE	0x00000010	/**< This video mode may be resized */
 #define SDL_NOFRAME	0x00000020	/**< No window caption or edge frame */
+#define SDL_SWIZZLEBGR	0x00000040	/**< Video mode has BGR subpixel ordering (not RGB) */
```
Comment (JohnnyonFlame) [inline include/SDL_video.h:147]: Support for both this and SDL_TRIPLEBUF extensions can be tested with [the following test app.](https://gist.github.com/JohnnyonFlame/a92a551e49bbd1d8d83b024ee7288652)

File: src/video/kmsdrm/SDL_kmsdrmvideo.c [hunk 3]
```diff
@@ -278,8 +317,7 @@ int KMSDRM_VideoInit(_THIS, SDL_PixelFormat *vformat)
 
 SDL_Rect **KMSDRM_ListModes(_THIS, SDL_PixelFormat *format, Uint32 flags)
 {
-	/** TODO:: Unstub. **/
-   	return (SDL_Rect **) -1;
+	return drm_vid_modes;
```
Comment (JohnnyonFlame) [inline src/video/kmsdrm/SDL_kmsdrmvideo.c:288]: For the user to be able to set arbitrary, non-native resolutions and let DRM sort it out, it's necessary to hack around SDL_SetVideoMode to attempt setting modes that are unlisted here.

This is necessary for applications that won't respect -1 being "any mode is supported".

**Comment by pcercuei** [comment]
Awesome work! Kudos!

**Pull request merged**